include/getopt.h

```diff
@@ -0,0 +1,154 @@
+
+#include <nuttx/config.h>
+#include <nuttx/compiler.h>
+
+#include <unistd.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Names for the values of the `has_arg' field of `struct option'. */
+
+#define no_argument             0
+#define required_argument       1
+#define optional_argument       2
+
+#define NO_ARG                  no_argument
+#define REQUIRED_ARG            required_argument
+#define OPTIONAL_ARG            optional_argument
+
+/* The GETOPT_DATA_INITIALIZER macro is used to initialize a statically-
+ * allocated variable of type struct getopt_data.
+ */
+
+#define GETOPT_DATA_INITIALIZER {0, 0, 1, 0, 0, 0, 0}
+
+/****************************************************************************
+ * Public Type Definitions
+ ****************************************************************************/
+
+#undef EXTERN
+#if defined(__cplusplus)
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
+#endif
+
+/* Describe the long-named options requested by the application.
+ * The LONG_OPTIONS argument to getopt_long or getopt_long_only is a vector
+ * of `struct option' terminated by an element containing a name which is
+ * zero.
+ *
+ * The field `has_arg' is:
+ * no_argument       (or 0) if the option does not take an argument,
+ * required_argument (or 1) if the option requires an argument,
+ * optional_argument (or 2) if the option takes an optional argument.
+ *
+ * If the field `flag' is not NULL, it points to a variable that is set
+ * to the value given in the field `val' when the option is found, but
+ * left unchanged if the option is not found.
+ *
+ * To have a long-named option do something other than set an `int' to
+ * a compiled-in constant, such as set a value from `optarg', set the
+ * option's `flag' field to zero and its `val' field to a nonzero
+ * value (the equivalent single-letter option character, if there is
+ * one).  For long options that have a zero `flag' field, `getopt'
+ * returns the contents of the `val' field.
+ */
+
+struct option
+{
+  FAR const char *name; /* The name of the long option */
+  int has_arg;          /* One of the above macros */
+  FAR int *flag;        /* Determines if getopt_long() returns a
+                         * value for a long option; if it is
+                         * non-NULL, 0 is returned as a function
+                         * value and the value of val is stored in
+                         * the area pointed to by flag.  Otherwise,
+                         * val is returned.
+                         */
+  int val;              /* Determines the value to return if flag is
+                         * NULL.
+                         */
+};
+
+/* The getopt_data structure is for reentrancy. Its members are similar to
+ * the externally-defined variables.
+ */
+
+typedef struct getopt_data
+{
```

Comment:
Coding standard requires that type definition names end with _t

Reply:
Ok, getopt_data doesn't specify in any document. I will change it.